### ISSUE TITLE
MemoryError: generator cannot be allocated - Possibile Memory Leaks

### ISSUE THREAD

**Issue by awused** [seanegoodwin/PandaViewer#59]
2015-10-07 14:06:24,004: Program ERROR Got unhandled exception
Traceback (most recent call last):
  File "C:\PandaViewer\PandaViewer\Threads.py", line 69, in run
  File "C:\PandaViewer\PandaViewer\Threads.py", line 343, in _run
  File "C:\PandaViewer\PandaViewer\Program.py", line 375, in filter_galleries
MemoryError: generator cannot be allocated
2015-10-07 14:06:25,006: Program ERROR Got unhandled exception
Traceback (most recent call last):
  File "C:\PandaViewer\PandaViewer\Threads.py", line 69, in run
  File "C:\PandaViewer\PandaViewer\Threads.py", line 343, in _run
  File "C:\PandaViewer\PandaViewer\Program.py", line 375, in filter_galleries
MemoryError: generator cannot be allocated

At the point it crashed it had been well over 16 hours since I took any actions at all in the program. I was not watching the application but it looks like it could be a memory leak. I'm going to watch it over time to see if I can catch it leaking.

However keeping an eye on the memory usage I have noticed the memory usage grows permanently after viewing a gallery's info (with all of its thumbnails). Repeatedly viewing the same large tankouban causes what looks very much like a leak; each time I view it about 50MB is allocated but only ~40MB is freed when I leave. Viewing it ten times causes the program to use roughly 100MB more memory than it was using before which is twice as much as it uses when actively viewing the gallery.

**Comment by seanegoodwin** [comment]
There definitely seems to be a memory leak. The interesting part is that it only seems to happen when running the exe, and not from source. Which means it probably isn't an issue with PV.

I'm probably going to change the tool I use to distribute PV and see if it helps.
Also, just as a note - if you do something (like open the info page) and then 'undo' that (by leaving the page), the memory isn't necessarily going to go down to the same level, at least not immediately.

**Comment by awused** [comment]
Yeah I know it's not necessarily going to drop down immediately, not with a garbage collected language, but a consistent ~10MB net increase each time I view the same gallery is way more than I'd have expected.

I didn't catch any gradual memory leak but it did crash again today while I was at work.

**Comment by seanegoodwin** [comment]
Try with the latest version. It seems to be working a lot better on my end.
